composer/src/main/kotlin/com/gojuno/composer/Args.kt

```diff
@@ -92,9 +94,24 @@ private class JCommanderArgs {
             names = arrayOf("--devices"),
             required = false,
             variableArity = true,
-            description = "Connected devices/emulators that will be used to run tests against. If not passed — tests will run on all connected devices/emulators. Usage example: `--devices emulator-5554 emulator-5556`."
+            description = "Connected devices/emulators that will be used to run tests against. If not passed — tests will run on all connected devices/emulators. Can't pass `--devices` with `--device-pattern`. Usage example: `--devices emulator-5554 emulator-5556`."
     )
     var devices: List<String>? = null
+
+    @Parameter(
+            names = arrayOf("--device-pattern"),
+            required = false,
+            description = "Connected devices/emulators that will be used to run tests against. If not passed — tests will run on all connected devices/emulators. Can't pass `--device-pattern` with `--devices`. Usage example: `--device-pattern \"somePatterns\"`."
+    )
+    var devicePattern: String? = null
+}
+
+private fun validateArguments(args: Args): Args{
+    if(!args.devicePattern.isEmpty() && !args.devices.isEmpty()) {
+        throw ParameterException("Can't pass both --devices and --device-pattern at the same time.")
```

Comment:
`IllegalArgumentException` also looks appropriate, used by JDK and in general is more well-known for developers :)